oracle/src/mevSender.js

```diff
@@ -0,0 +1,159 @@
+require('../env')
+const path = require('path')
+const BigNumber = require('bignumber.js')
+const { redis } = require('./services/redisClient')
+const logger = require('./services/logger')
+const { sendTx } = require('./tx/sendTx')
+const { getNonce, getChainId, getBlock } = require('./tx/web3')
+const { addExtraGas, checkHTTPS, watchdog } = require('./utils/utils')
+const { EXIT_CODES, EXTRA_GAS_PERCENTAGE, MAX_GAS_LIMIT } = require('./utils/constants')
+const { estimateProfit } = require('./events/processAMBCollectedSignaturesMEV')
+
+if (process.argv.length < 3) {
+  logger.error('Please check the number of arguments, config file was not provided')
+  process.exit(EXIT_CODES.GENERAL_ERROR)
+}
+
+const config = require(path.join('../config/', process.argv[2]))
+
+const { web3, mevForeign, validatorAddress } = config
+
+let chainId = 0
+let flashbotsProvider
+
+async function initialize() {
+  try {
+    const checkHttps = checkHTTPS(process.env.ORACLE_ALLOW_HTTP_FOR_RPC, logger)
+
+    web3.currentProvider.urls.forEach(checkHttps(config.id))
+
+    chainId = await getChainId(web3)
+    flashbotsProvider = await mevForeign.getFlashbotsProvider(chainId)
+    return runMain()
+  } catch (e) {
+    logger.error(e.message)
+    process.exit(EXIT_CODES.GENERAL_ERROR)
+  }
+}
+
+async function runMain() {
+  try {
+    if (redis.status === 'ready') {
+      if (config.maxProcessingTime) {
+        await watchdog(main, config.maxProcessingTime, () => {
+          logger.fatal('Max processing time reached')
+          process.exit(EXIT_CODES.MAX_TIME_REACHED)
+        })
+      } else {
+        await main()
+      }
+    }
+  } catch (e) {
+    logger.error(e)
+  }
+
+  setTimeout(runMain, config.pollingInterval)
+}
+
+async function main() {
+  try {
+    const jobs = Object.values(await redis.hgetall(config.mevJobsRedisKey)).map(JSON.parse)
+    const totalJobs = jobs.length
+
+    if (totalJobs === 0) {
+      logger.info('Nothing to process')
```

Comment:
Seems that this message will flood the logs. Does it make sense to make use the "debug" level for it?

Reply:
Ok, no problem